### ISSUE TITLE
Is it possible to specify which GPU is selected to run the code?

### ISSUE THREAD

**Issue by mycrazycracy** [fchollet/keras#6959]
Hey, I don't know if it is possible to specify the GPU number when I want to play something with NN (training or prediction). And I may use different GPUs for different parts of my code.
I ask this question because I want to use Keras to train a network with triplet loss. And I want one GPUs to do the BP. At the same time, I want to use another GPU to do forward and help me to choose the "hardest" positive or negative pair. If I could use multiple GPUs, I think the training procedure would significantly speed up. 
I do know in tensorflow, it is possible to do so using "with device", but I use theano now and I wonder if it is possible to simulate this behavior just using the Keras wrapper?

**Comment by kgrm** [comment]
Before you import keras, 

`import os; os.environ["THEANO_FLAGS"] = "device=gpuX` ,

where `X` is the GPU index.

**Comment by mycrazycracy** [comment]
hey, @kgrm thanks for your advice. I know how to specify the gpu globally. But actually my problem is: when I have two GPUs, I want to use the different one for different part of my code.
That is I want to use gpu0 for the training and gpu1 for the data generation.
I don't know whether it is clear or not?

**Comment by kgrm** [comment]
I'm not sure whether this can be done in one script, but what you could do is have one script for data generation and one for training, and have them pass messages between, e.g. with sockets or shared i/o streams.
